spec/controllers/wishlists_controller_spec.rb

```diff
@@ -1,84 +1,124 @@
+  let(:invalid_params) { { name: "", description: "" } }
 
-  describe "GET #new" do
-    before do
-      request_get
-    end
+  context "when user is logged in" do
+    sign_in_user
 
-    it "returns 200 HTTP status code" do
-      expect(response).to have_http_status(200)
-    end
+    describe "GET #index" do
+      before(:each) { get :index }
 
-    it "renders the 'new' template" do
-      expect(response).to render_template(:new)
-    end
+      it "returns 200 HTTP status code" do
+        expect(response).to have_http_status(:ok)
+      end
 
-    it "assigns a new wishlist as @wishlist" do
-      expect(assigns(:wishlist)).to be_a_new(Wishlist)
+      it "renders the 'index' template" do
+        expect(response).to render_template(:index)
+      end
     end
-  end
 
-  describe "POST #create" do
-    context "with valid parameters" do
-      before do
-        request_post(valid_parameters)
+    describe "GET #new" do
+      before(:each) { get :new }
+
+      it "returns 200 HTTP status code" do
+        expect(response).to have_http_status(200)
       end
 
-      it "creates a new Wishlist" do
-        expect do
-          post :create, params: { wishlist: valid_parameters }
-        end.to change(Wishlist, :count).by(1)
+      it "renders the 'new' template" do
+        expect(response).to render_template(:new)
       end
 
-      it "assigns a newly created wishlist as @wishlist" do
-        expect(assigns(:wishlist)).to be_a(Wishlist)
+      it "assigns a new wishlist as @wishlist" do
+        expect(assigns(:wishlist)).to be_a_new(Wishlist)
       end
+    end
+
+    describe "POST #create" do
+      context "with valid parameters" do
+        before(:each) { post_create(valid_params) }
+
+        it "creates a new Wishlist" do
+          expect { post :create, params: { wishlist: valid_params } }.to change(Wishlist, :count).by(1)
+        end
 
-      it "checks if assigned wishlist as @wishlist is persisted" do
-        expect(assigns(:wishlist)).to be_persisted
+        it "assigns a newly created wishlist as @wishlist" do
+          expect(assigns(:wishlist)).to be_a(Wishlist)
+        end
+
+        it "checks if assigned wishlist as @wishlist is persisted" do
+          expect(assigns(:wishlist)).to be_persisted
+        end
+
+        it "redirects to the wishlists index page" do
+          expect(response).to redirect_to(wishlists_path)
+        end
       end
 
-      it "redirects to the form page" do
-        expect(response).to redirect_to(new_wishlist_path)
+      context "with invalid parameters" do
+        before(:each) { post_create(invalid_params) }
+
+        it "returns 200 HTTP status code" do
+          expect(response).to have_http_status(200)
+        end
+
+        it "assigns a newly created but unsaved wishlist as @wishlist" do
+          expect(assigns(:wishlist)).to be_a_new(Wishlist)
+        end
+
+        it "doesn't create new wishlist" do
```

Comment:
It's a little bit confusing title comparing it with one above. Maybe sth like "doesn't save new wishlist to a database"?